### PR TITLE
add missing MSIN field to simcards

### PR DESCRIPTION
**Pull request by btry**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | 
| Fixed tickets |

MSIN is the last 8 or 10 digits of IMSI
IMSI is the concatenation of:
- MCC (Mobile Country Code) (3 digits)
- MNC (Mobile Network Code) (2 or 3 digits)
- MSIN (Mobile Subscriber Identification Number) (8 or 10 digits)

Having 3 digits length MCC and 3 digits length MNC and 10 digits length MSIN is not allowed
because the full length is at most 15 digits.

Possible combinations are:
3 digits MCC + 2 digits MNC + 10 igits MSIN = 15 digits
3 digits MCC + 3 ditigs MNC + 8 digits MSIN = 14 digits
3 digits MCC + 2 digits MNC + 8 digits MSIN = 13 digits

MCC and MNC are stored in the line operator itemtype
MSIN is stored in a items_simcards itemtype. To save the full IMSI, one must
create a simcard component, instanciate it, fill its MSIN field and associate it to a line, associate the line to an operator with MCC and MNC fields populated.

Also fix wrong base class for line itemtype, preventing to create a line from items_simcards form

### CODE REVIEW

File: inc/item_devices.class.php [hunk 1]
```diff
@@ -1109,20 +1107,42 @@ function copyToClipboard(item) {
...
+               case 'dropdown':
+                  $fieldType = 'dropdown';
+                  break;
+
+               default:
+                  $fieldType = 'textfield';
+            }
+            echo "<label for='${fieldType}_$field$rand'>".$attributs['long name']."</label>";
+         } else {
+            echo $attributs['long name'];
```
Comment (btry) [inline inc/item_devices.class.php]: if no right to read the field, do not add a LABEL  tag

File: inc/item_devices.class.php [hunk 2]
```diff
@@ -1109,20 +1107,42 @@ function copyToClipboard(item) {
...
+         }
+         echo "</td><td>";
+
```
Comment (btry) [inline inc/item_devices.class.php]: @trasher you asked whine I was working on this code to add html LABEL tags. here it is

File: inc/item_devices.class.php [hunk 3]
```diff
@@ -1109,20 +1107,42 @@ function copyToClipboard(item) {
...
          // Do the field needs a user action to display ?
          if (isset($attributs['protected']) && $attributs['protected']) {
             $protected = true;
          } else {
             $protected = false;
          }
 
-         if (!isset($attributs['datatype'])) {
-            $attributs['datatype'] = 'text';
```
Comment (btry) [inline inc/item_devices.class.php]: moved to L1110

File: inc/item_devices.class.php [hunk 4]
```diff
@@ -1142,10 +1162,19 @@ function copyToClipboard(item) {
...
+               $options_tooltip = ['contentid' => $comment_id,
+                  'linkid'    => $link_id,
+                  'display'   => false];
+               $out.= '&nbsp;'.Html::showToolTip($tooltip, $options_tooltip);
+            }
             if ($protected) {
                $out.= '<i class="fa fa-eye pointer disclose" ';
                $out.= 'onmousedown="showField(\'' . $field . $rand . '\')" ';
-               $out.= 'onmouseup="hideField(\'' . $field . $rand . '\')"></i>';
+               $out.= 'onmouseup="hideField(\'' . $field . $rand . '\')" ';
```
Comment (trasher) [inline inc/item_devices.class.php]: Guess there is a merge issue here

**Comment by btry** [comment]
@trasher 

I think the commit you added may introduce an issue.

I found I need to hide the protected field when toue user releases the button of his mouse, and also when the mouse is pressed and the user moves out of the icon.

Then I end up to handle 3 events, not only two.

**Comment by trasher** [comment]
OK, dropped :)

**Comment by btry** [comment]
@orthagh @trasher : tests passed now

**Pull request merged**